### ISSUE TITLE
[KED-973] HDFS3DataSet won't save a dataframe in table mode (Update)

### ISSUE THREAD

**Issue by 6Hhcy** [quantumblacklabs/kedro#81]
## Description
I have a dataframe with categorical variables. 

_save() doesn't work using this dataset, since it is being "put" rather than appended to the store

(see this issue in Pandas: https://github.com/pandas-dev/pandas/issues/11843)

I recommend using this:

```
store.append(key = self._key, value=data, append=False)
```
rather than:
```
store[self._key] = data
```
source: kedro/io/hdf_s3.py - line 166:

## Update
Subsequent to applying this patch, I found that attempting to save H5 files of size = 24mb caused a new error (note: small files are ok):

```
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\kedro\io\core.py", line 232, in save
    self._save(data)
  File "C:\Users\Me\PycharmProjects\MyProj\datascience_kedro\src\my_proj\nodes\custom_datasets.py", line 120, in _save
    store.append(key=self._key, value=data, append=False)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\pandas\io\pytables.py", line 987, in append
    **kwargs)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\pandas\io\pytables.py", line 1370, in _write_to_group
    s.create_index(columns=index)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\pandas\io\pytables.py", line 3365, in create_index
    v.create_index(**kw)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\table.py", line 3609, in create_index
    tmp_dir, _blocksizes, _verbose)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\table.py", line 319, in _column__create_index
    blocksizes=blocksizes)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\index.py", line 400, in __init__
    super(Index, self).__init__(parentnode, name, title, new, filters)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\group.py", line 238, in __init__
    super(Group, self).__init__(parentnode, name, _log)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\node.py", line 275, in __init__
    self._g_post_init_hook()
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\index.py", line 554, in _g_post_init_hook
    self.create_temp()
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\site-packages\tables\index.py", line 1013, in create_temp
    ".tmp", "pytables-", self.tmp_dir)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\tempfile.py", line 344, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "C:\Users\Me\Miniconda3\envs\datasci\lib\tempfile.py", line 262, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
FileNotFoundError: [Errno 2] No such file or directory: 'data/\\pytables-_0u2doys.tmp'

```
This seems to occur while attempting to create an index on the H5 file. The intermediate tempfile in the S3 bucket is not created (note this error happens in both Win10 and Ubuntu 18.04.3 LTS, so I don't think it's an improperly formed path). 

My quick hack for this is to go into: `/site-packages/pandas/io/pytables.py`
in the `create_index` method and set `optlevel=False` -> `optlevel=0`. This prevents index optimisation and hence no attempt is made  to create a tempfile.

## Context
How has this bug affected you? What were you trying to accomplish?

## Steps to Reproduce
1. [First Step]
2. [Second Step]
3. [and so on...]

## Expected Result
Tell us what should happen.

## Actual Result
Tell us what happens instead.

```
-- If you received an error, place it here.
```

```
-- Separate them if you have more than one.
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used 0.14.3
* Python version used 3.6.6:
* Operating system and version Win10 and Ubuntu 18.04.3 LTS

**Comment by DmitriiDeriabinQB** [comment]
@6Hhcy thank you for reporting this! We will need to do further investigation of the issue. We will get back to you as soon as we can. In the meantime, please feel free to test your suggestions and submit a PR with the proposed fixes if you would like to.

**Comment by andrii-ivaniuk** [comment]
@6Hhcy I was able to reproduce the issue with saving the dataframe with categorical variables, so it will be fixed in the upcoming release.
However, I failed to reproduce the second one. I manage to save 38Mb to s3.

**Comment by andrii-ivaniuk** [comment]
The issue with saving the dataframe with categorical variables was fixed and will be available in the next Kedro release.

**Comment by DmitriiDeriabinQB** [comment]
Fixed in d06c2b8370368021f2a7295eae508d0c417842cd
